todo_app/templates/boardtasks.html

```diff
@@ -0,0 +1,57 @@
+{% extends "layout.html" %}
+{% block title %}Andy's To-Do App{% endblock %}
+
+{% block content %}
+
+  <div class="jumbotron">
+    <h1 class="display-4">Andy's To-Do App</h1>
+    <p class="lead">Just another to-do app.</p>
+  </div>
+  <div class="row">
+    <div class="col-md-12">
+      <h2 style="color:#FF0000">Things To Do</h2>
+      <ul class="list-group mb-4">   
+        {% for card in todo_cards %}
+        <li style="color:#FF0000"> {{ card }} </li>
+          <li><a href="/start/{{ card }}" class="btn btn-sm btn-success">Doing</a></li>       
+        {% endfor %}
+        <br> 
+        <h2 style="color:#FD6A02">Doing</h2>
+      <ul class="list-group mb-4">   
+        {% for card in doing_cards %}
+         <li style="color:#FD6A02"> {{ card }} </li>
+         <li><a href="/redo_doing/{{ card }}" class="btn btn-sm btn-success">Re-Do</a>           <a  href="/complete/{{ card }}" class="btn btn-sm btn-success">Done</a></li>          
+        {% endfor %}
+        <br>  
+        <h2 style="color:#228B22">Done</h2>
+      <ul class="list-group mb-4">   
+        {% for card in done_cards.keys() %}
+         <li style="color:#228B22"> {{ card }} </li>
+         <li><a href="/restart/{{ card }}" class="btn btn-sm btn-success">Re-Do</a>           <a  href="/redo/{{ card }}" class="btn btn-sm btn-success">Still Doing</a></li>          
+        {% endfor %}
+        <br>         
+      </ul>
+  <div class="row">
+    <div class="col-md-12">
+      
+      <form action="/" method="get">
+        <button type="submit"> Click here for to return to board index </button>
+      </form>
+<br>
+<h2>Create a New Task</h2>
+  <form action="" method="post" class="form-example">
+    <div class="form-example">
+       <label for="title">Enter title of task  : </label>
+       <input type="text" name="title" id="title" required>
+       <select name="idList">
+    {% for name in lists.keys() %}
+        <option value="{{ lists[name] }}">{{ name }}</option>
+    {% endfor %}
+      </select>
+    <div class="form-example">
+       <input type="submit" value="Add Task">
+     </div>
+  </form>
+</div>
```

Comment:
My rewrite:

There are a few changes, but in particular:
- You still have some "get" forms that could just be `<a>` tags as they don't have any input
- I use ids for them for simplicity as the urls are not user visible
- You have some tags that don't close and some confusing usage of <li> - each li should normally be equivalent rather than alternating between a card name and a row of buttons.
- Not sure how you would want to generate the text for the sections programmatically for arbitrary lists, so it's still just displaying the first three lists in order.

```<div class="jumbotron">
    <h1 class="display-4">Andy's To-Do App</h1>
    <p class="lead">Just another to-do app.</p>
  </div>
  <div class="row">
    <div class="col-md-12">
      <h2 style="color:#FF0000">Things To Do</h2>
      <ul class="list-group mb-4">
        {% for card in lists[0]['cards'] %}
        <li style="color:#FF0000">
          {{ card.name }} <br>
          <a href="{{ url_for('move_card', card_id=card.id, list_id=lists[1].id) }}" class="btn btn-sm btn-success">Doing</a>
        </li>     
        {% endfor %}
      </ul>
      <h2 style="color:#FD6A02">Doing</h2>
      <ul class="list-group mb-4">   
        {% for card in lists[1]['cards'] %}
          <li style="color:#FD6A02">
            {{ card.name }} <br>
            <a href="{{ url_for('move_card', card_id=card.id, list_id=lists[0].id) }}" class="btn btn-sm btn-success">Re-Do</a>
            <a href="{{ url_for('move_card', card_id=card.id, list_id=lists[2].id) }}" class="btn btn-sm btn-success">Done</a>
          </li>          
        {% endfor %}
      </ul>
      <h2 style="color:#228B22">Done</h2>
      <ul class="list-group mb-4">   
        {% for card in lists[2]['cards'] %}
          <li style="color:#228B22">
            {{ card.name }} <br>
            <a href="{{ url_for('move_card', card_id=card.id, list_id=lists[0].id) }}" class="btn btn-sm btn-success">Re-Do</a>
            <a href="{{ url_for('move_card', card_id=card.id, list_id=lists[1].id) }}" class="btn btn-sm btn-success">Still Doing</a>
          </li>
        {% endfor %}
      </ul>
  <div class="row">
    <div class="col-md-12">
      <a href="/"> Click here to return to boards list </a>
    </div>
  </div>
  <h2>Create a New Task</h2>
  <form action="{{ url_for('create_board_task', board_name=board_name) }}" method="post" class="form-example">
    <div class="form-example">
      <label for="title">Enter title of task  : </label>
      <input type="text" name="title" id="title" required>
      <select name="idList">
        {% for list in lists %}
          <option value="{{ list.id }}">{{ list.name }}</option>
        {% endfor %}
      </select>
    <div class="form-example">
      <input type="submit" value="Add Task">
    </div>
  </form>
```